build.sh

```diff
@@ -251,17 +263,27 @@ build_mscorlib()
 
     # The cross build generates a crossgen with the target architecture.
     if [ $__CrossBuild != 1 ]; then
-       if [ $__SkipCoreCLR == 0 -a -e $__BinDir/crossgen ]; then
-           echo "Generating native image for mscorlib."
-           $__BinDir/crossgen $__BinDir/mscorlib.dll
-           if [ $? -ne 0 ]; then
-               echo "Failed to generate native image for mscorlib."
-               exit 1
-           fi
+       # The architecture of host pc must be same architecture with target.
+       if [[ ( "$__HostArch" == "x64" ) && ( "$__BuildArch" == "x64" ) ]]; then
```

Comment:
I would be simpler and easier to follow if there is the simple exact match first, and then the additional mismatched combinations you would like to allow after it. Something like:

```
if [[ $__HostArch" == "$__BuildArch" ]];
    build_mscorlib_ni;
elif [[ ( "$__HostArch" == "x64" ) && ( "$__BuildArch" == "x86" ) ]]; 
    build_mscorlib_ni;
elif [[ ( "$__HostArch" == "arm64" ) && ( "$__BuildArch" == "arm" ) ]]; 
    build_mscorlib_ni;
```